programs/vest/src/vest_processor.rs

```diff
@@ -97,25 +97,25 @@ pub fn process_instruction(
         VestInstruction::InitializeAccount { .. } => {}
         VestInstruction::SetTerminator(new_pubkey) => {
             verify_signed_account(
-                next_keyed_account(keyed_accounts_iter)?,
+                keyed_account_at_index(keyed_accounts, 1)?,
```

Comment:
This looks fragile compared to the previous code. Why not remove one account at a time like it was?

Reply:
Just read the PR description.  I'll like to spend some time understanding the "borrow collision" more.

Reply:
Iterators won't do as they keep an internal reference which would collide with the one in `invoke_context`.
See my comments on that in the header / description of the PR.